### PR TITLE
Added a Maven build

### PR DESCRIPTION
**Pull request by adamretter**
Hey Dave,

This might be of interest to you or not? I wanted to have a look at your code and be able to build it myself, so I added a Maven build in this PR. Once you have installed Apache Maven, you can now just run:

```
git clone https://github.com/blooprint/whiteSocket.git
cd whiteSocket
mvn clean compile assembly
java -jar target/whiteSocket-0.1.4-SNAPSHOT-uber.jar
```

It will also produce an "app" for you with scripts for running on Windows and Mac/Linux/Unix, just run:
```
mvn clean compile package appassembler:assemble
```
You can then find the app in `target/appassembler`, it's the sort of thing you could zip up and distribute.

Anyway, might be useful for you. Also the Maven build is natively compatible with IntelliJ IDE (or NetBeans) if you use that. If you use Eclipse I think it also has Maven support but I am not familiar with it.

### CODE REVIEW

**Comment by ddaaggeett** [comment]
This is helpful. Maven is something I haven't used willingly, though I'm sure I've used it without knowing.

I'll to dig into this to understand what you're saying and I'm sure I won't find a reason not to merge.

Forgive my lack of knowledge - am I correct in assuming this will make dev adaptation to building the whiteSocket more seamless?

Also, you made my day!  FIRST PR - WOOT WOOOOOT!!!!!

**Comment by adamretter** [comment]
I am not quite sure what you are asking there, what do you mean by "dev adaptation"?

**Comment by ddaaggeett** [comment]
Will it make NetBeans and IntelliJ able to build? It's a dumb question. You already answered it in the original post.

**Comment by adamretter** [comment]
Yes they should just work with your project now.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,257 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.sonatype.oss</groupId>
```
Comment (ddaaggeett) [inline pom.xml:8]: @adamretter - is the [deprecation](https://github.com/sonatype/oss-parents) of this something to worry about?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    </parent>
+
+    <groupId>xyz.blooprint</groupId>
+    <artifactId>whiteSocket</artifactId>
+    <version>0.1.4-SNAPSHOT</version>
+
+    <name>whiteSocket</name>
+    <description>Generates diffs between images</description>
+    <url>http://https://github.com/blooprint/whiteSocket/</url>
+    <inceptionYear>2016</inceptionYear>
```
Comment (ddaaggeett) [inline pom.xml]: 2015

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+    </licenses>
+
+    <scm>
+        <connection>scm:git:https://github.com/blooprint/whiteSocket.git</connection>
+        <developerConnection>scm:git:https://github.com/blooprint/whiteSocket.git</developerConnection>
+        <url>scm:git:https://github.com/blooprint/whiteSocket.git</url>
+    </scm>
+
+    <properties>
+        <java.version>1.7</java.version>
```
Comment (ddaaggeett) [inline pom.xml]: 1.8

**Review by ddaaggeett — CHANGES_REQUESTED** [review]
github is becoming more and more fascinating. i just have a few considerations before moving forward.

File: pom.xml (hunk 1 above)
Comment (adamretter) [reply pom.xml:8]: @ddaaggeett Hah I didn't even know it was deprecated! So nope I haven't worried about it. It seems that the alternative now is just to pull some of the settings from their pom.xml directly into your own pom.xml. I think its probably not worth worrying about just yet, can always improve that later...

File: pom.xml (hunk 2 above)
Comment (adamretter) [reply pom.xml]: I will fix that...

File: pom.xml (hunk 3 above)
Comment (adamretter) [reply pom.xml]: I will fix that...

**Comment by adamretter** [comment]
@ddaaggeett Okay I fixed those two issues and rebased the commit in the PR for you.

File: pom.xml (hunk 1 above)
Comment (ddaaggeett) [reply pom.xml:8]: sounds good

**Comment by ddaaggeett** [comment]
@adamretter 
Adam, thanks for the first pull request (ignoring my personal a while back I did just for simple testing).
I tested - it runs - awesome! This was great to play with some branching and learn about some Maven stuff. This will definitely make it better for Java developers to come aboard and use their IDE of choice. CHEERS!

**Pull request merged**